src/main/java/net/dv8tion/jda/api/interactions/components/ItemComponent.java

```diff
@@ -33,4 +33,26 @@ default int getMaxPerRow()
     {
         return getType().getMaxPerRow();
     }
+
+    /**
+     * Whether this component can be used in {@link Modal Modals}.
+     *
+     * @return Whether this component can be used in Modals.
+     *
+     * @see    Type#isModalCompatible()
+     */
+    default boolean isModalCompatible()
+    {
+        return getType().isModalCompatible();
+    }
+
+    /**
+     * Whether this component can be used in {@link net.dv8tion.jda.api.entities.Message Messages}.
+     *
+     * @return Whether this component can be used in Messages.
+     */
+    default boolean isMessageCompatible()
+    {
+        return getType().isMessageCompatible();
+    }
```

Comment:
Why are these defined on on ItemComponent and LayoutComponent instead of the Component class?

Reply:
Because their behaviour is different. For `ItemComponent`s it logically only checks the component itself, while on `LayoutComponent` it checks whether both the layout is compatible AND any of the components inside